### PR TITLE
Functional tests - Add test 'Check invoice downloaded from orders list'

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add functional test 'Check invoice downloaded from orders list'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | `DOWNLOAD_PATH='downloadPath' TEST_PATH="functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js" URL_FO=shopURL/ npm run specific-test`

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18306)

### CODE REVIEW

File: tests/puppeteer/campaigns/data/demo/orderStatuses.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-    paymentAccepted: {index: 7, status: 'Payment accepted'},
-    paymentError: {index: 8, status: 'Payment error'},
-    awaitingBankWire: {index: 10, status: 'Awaiting bank wire payment'},
-    shipped: {index: 14, status: 'Shipped'},
+    awaitingCheckPayment: {id: 1, status: 'Awaiting check payment'},
+    paymentAccepted: {id: 2, status: 'Payment accepted'},
+    shipped: {id: 4, status: 'Shipped'},
+    canceled: {id: 6, status: 'Canceled'},
+    paymentError: {id: 8, status: 'Payment error'},
+    awaitingBankWire: {id: 10, status: 'Awaiting bank wire payment'},
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/data/demo/orderStatuses.js:8]: What happened here ?

File: tests/puppeteer/campaigns/functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    homePage: new HomePage(page),
+    productPage: new ProductPage(page),
+    cartPage: new CartPage(page),
+    checkoutPage: new CheckoutPage(page),
+    orderConfirmationPage: new OrderConfirmationPage(page),
+  };
+};
+
+/*
+Create order in FO win bank wire payment
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js]:
```suggestion
Create order in FO with bank wire payment
```

File: tests/puppeteer/campaigns/functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      await expect(referenceExist, `Reference '${orderInformation.reference}' does not exist in invoice`).to.be.true;
+      // Check country name in delivery Address in pdf
+      const deliveryExist = await files.checkTextInPDF(invoiceFilename, orderInformation.delivery);
+      await expect(deliveryExist, `Country name '${orderInformation.delivery}' does not exist in invoice`).to.be.true;
+      // Check customer name in pdf
+      const customerExist = await files.checkTextInPDF(invoiceFilename, orderInformation.customer.slice(3));
+      await expect(customerExist, `Country name '${orderInformation.customer}' does not exist in invoice`).to.be.true;
+      // Check total paid in pdf
+      const totalPaidExist = await files.checkTextInPDF(invoiceFilename, orderInformation.totalPaid);
+      await expect(totalPaidExist, `Total paid '${orderInformation.totalPaid}' does not exist in invoice`).to.be.true;
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js:146]: We won't have the old problem with the return to next line ?

**Review by SimonGrn — COMMENTED** [review]
thanks @boubkerbribri

File: tests/puppeteer/campaigns/data/demo/orderStatuses.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/data/demo/orderStatuses.js:8]: Index was not used at all and it was referred to the position of status in old orders page. 
So indexes is now ids, and used in function setOrderStatus.

File: tests/puppeteer/campaigns/functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js (hunk 3 above)
Comment (boubkerbribri) [reply tests/puppeteer/campaigns/functional/BO/02_orders/01_orders/04_checkInvoiceDownloadedFromList.js:146]: I run `npm run lint` before pushing the PR, so i don't think so

**Review by SimonGrn — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Pull request merged**